cryptoprotocol/src/main/java/de/upb/crypto/incentive/cryptoprotocol/model/Token.java

```diff
@@ -1,4 +1,156 @@
 package de.upb.crypto.incentive.cryptoprotocol.model;
 
-public class Token {
+
+import de.upb.crypto.craco.common.GroupElementPlainText;
+import de.upb.crypto.craco.sig.interfaces.VerificationKey;
+import de.upb.crypto.craco.sig.sps.eq.SPSEQSignature;
+import de.upb.crypto.craco.sig.sps.eq.SPSEQSignatureScheme;
+import de.upb.crypto.incentive.cryptoprotocol.exceptions.SPSEQException;
+import de.upb.crypto.incentive.cryptoprotocol.model.keys.provider.ProviderPublicKey;
+import de.upb.crypto.math.interfaces.structures.GroupElement;
+import de.upb.crypto.math.serialization.Representable;
+import de.upb.crypto.math.serialization.Representation;
+import de.upb.crypto.math.serialization.annotations.v2.ReprUtil;
+import de.upb.crypto.math.serialization.annotations.v2.Represented;
+import de.upb.crypto.math.structures.cartesian.GroupElementVector;
+import de.upb.crypto.math.structures.zn.Zn;
+import de.upb.crypto.math.structures.zn.Zn.ZnElement;
+
+import java.net.PortUnreachableException;
+
+
+/**
+ * class representing a token from a mathematical point of view (meaning: as a bunch of group elements and exponents modulo p from which you can compute a Pedersen commitment)
```

Comment:
This is a data holding class or? Maybe make this more clear in the java doc without the mathematical spin in the text ;-).

Reply:
Have to agree, this is a pretty odd docstring.
Cleaned it up a bit.